src/tools/debugger/session/Breakpoints.java

```diff
@@ -195,6 +203,33 @@ public Breakpoint getAsyncMessageRcvBreakpoint(final URI sourceUri,
     return bp;
   }
 
+  public synchronized void saveReceiverBreakpoint(final URI sourceUri, final int startLine,
+      final int startColumn, final int charLength) {
+    SectionBreakpoint bId = new SectionBreakpoint(sourceUri, startLine, startColumn, charLength);
+    if (!receiverBreakpoints.keySet().contains(bId)) {
+      this.receiverBreakpoints.put(bId, null);
+    }
+  }
+
+  public synchronized void updateReceiverBreakpoint(final BreakpointId bId, final Breakpoint bReceiver) {
+    if (receiverBreakpoints.keySet().contains(bId)) {
```

Comment:
should that `if` better be an assert? I'd assume there is something broken if that condition does not hold, no?